pyrs/core/summary_generator_stress.py

```diff
@@ -0,0 +1,200 @@
+"""
+This module generates reduction summary for stress in plain text CSV file
+"""
+from typing import Optional
+
+from pyrs.dataobjects.fields import StressField
+from pyrs.dataobjects.fields import StrainField
+from builtins import int, isinstance
+from pyrs.peaks.peak_collection import PeakCollection
+import math
+
+
+class SummaryGeneratorStress:
+    """
+        Generates a CSV summary from stress fields inputs from multiple project files on a grid point basis.
+    """
+
+    directions = ['11', '22', '33']
+    fields_3dir = ['d', 'FWHM', 'Peak_Height', 'Strain', 'Stress']
+
+    def __init__(self, filename: str, stress: StressField):
+        """Initialization
+
+        Parameters
+        ----------
+        filename: str
+            Name of the '.csv' file to write
+        stress: StressField
+            stress field input to generate summary or full csv files
+        """
+        # do file name checks
+        if not filename:
+            raise RuntimeError('Failed to supply output filename for Stress CSV output')
+
+        self._error_postfix = ' when creating stress file: ' + str(filename) + '\n'
+
+        if not filename.endswith('.csv'):
+            raise RuntimeError('File name must end with extension ".csv"' + self._error_postfix)
+
+        # check for length of lists
+        self._filename: str = str(filename)
+
+        if not stress:
+            raise RuntimeError(
+                'Error: stress of type StressField input can\'t be invalid' + self._error_postfix)
+
+        self._stress = stress
+
+        # check for filenames in StrainField per direction
+        for direction in SummaryGeneratorStress.directions:
+            strain = self._get_strain_field(direction)
+            assert isinstance(strain, StrainField)
+
+            # add exception if filenames is empty
+            if not strain.filenames:
+                raise RuntimeError('StrainField filenames in direction ' + str(direction) +
+                                   'can\'t be empty for Stress CSV output ' + self._filename)
+
+    def _write_csv_header(self, handle):
+        """
+          write projects names, peak_tags, Young's modulus and Poisson ratio
+        """
+        header = ''
+
+        for direction in SummaryGeneratorStress.directions:
+            strain = self._get_strain_field(direction)
+            assert isinstance(strain, StrainField)
+
+            line = '# Direction ' + str(direction) + ': '
+            for filename in strain.filenames:
+                run_number = filename[filename.index('HB2B_') + 5: filename.index('.h5')]
```

Comment:
should this check go in the backend `StrainField`, or is any file name is acceptable? Adding the check here will be fine with me, but if it's done somewhere else I'd rather not double-check.

Reply:
It should go into the backend, yes, provided the instrument scientists agree to abide by these names. I think there was some discussion about this because I vagely remember Pete confessing his love for short file names.

Reply:
I just put a question on Slack::Hydra::general about the file naming convention